releasenotes/notes/serializable-optimizers-642402828b0c733b.yaml

```diff
@@ -0,0 +1,29 @@
+---
+features:
+  - |
+    Enable dumping :class:`~qiskit.algorithms.optimizers.Optimizer` objects to JSON-serializable
+    dictionaries and load from these. 
```

Comment:
In addition to name changes (e.g. `settings` to `to_dict()`), I think we'd need to add a disclaimer here. The returned dictionary can be serialized to JSON only if the encoder they use supports all the types in it. For example, if `settings` contains a numpy array, which is not supported by the default `JSONEncoder `, then they need to use a custom encoder that supports it.

Reply:
good point, I updated the reno!